src/Servers/HttpSys/src/RequestProcessing/Request.cs

```diff
@@ -232,6 +238,59 @@ private AspNetCore.HttpSys.Internal.SocketAddress LocalEndPoint
 
         internal WindowsPrincipal User { get; }
 
+        public SslProtocols Protocol { get; private set; }
+
+        public CipherAlgorithmType CipherAlgorithm { get; private set; }
+
+        public int CipherStrength { get; private set; }
+
+        public HashAlgorithmType HashAlgorithm { get; private set; }
+
+        public int HashStrength { get; private set; }
+
+        public ExchangeAlgorithmType KeyExchangeAlgorithm { get; private set; }
+
+        public int KeyExchangeStrength { get; private set; }
+
+        private void GetTlsHandshakeResults()
+        {
+            var handshake = _nativeRequestContext.GetTlsHandshake();
+
+            Protocol = handshake.Protocol;
+            // https://docs.microsoft.com/en-us/windows/desktop/api/schannel/ns-schannel-_secpkgcontext_connectioninfo
+            // restore client/server bits so the result maps exaclty on SslProtocols constants
+            // Compare to https://referencesource.microsoft.com/#System/net/System/Net/SecureProtocols/_SslState.cs,8905d1bf17729de3
+#pragma warning disable CS0618 // Type or member is obsolete
+            if ((Protocol & SslProtocols.Ssl2) != 0)
```

Comment:
From what I can tell this is what is doing.
Ssl2 == 12, which is 1100 in binary. If the Protocol either has the 3rd or 4th bit set, set both bits. 

From refsource, it says
```
// restore client/server bits so the result maps exaclty on published constants
```

so for each ssl state, there is both a client and server set bit? 

No matter what this deserves a comment as a majority of the time, enums are not multibyte.

Reply:
The OS considers client and server TLS as different enum values (shown in the connectioninfo link above). .NET SslProtocols choose to combine those for some reason. I'll clarify the comment.